### PR TITLE
Ensure closing the operation log for batch submission in fast-failed case

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

This pull request fixes comment in PR [#5733](https://github.com/apache/kyuubi/pull/5733#issuecomment-1855055891)

## Describe Your Solution 🔧

To ensure the operation log closed even in fast-failed case.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by bowenliang123** [comment]
As #5854 will make sure the Operation closed by running `close` method, this PR is still necessary to ensure calling the close method in order to avoid escaping from the fast-failed case in batch submission. @pan3793 @turboFei

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala [hunk 1]
```diff
@@ -336,7 +335,7 @@ class BatchJobSubmission(
     }
   }
 
-  override def close(): Unit = withLockRequired {
+  override def close(): Unit = withLockRequired(withClosingOperationLog {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala:338]: nit: `(` => `{`

**Review by pan3793 — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala (hunk 1 above)
Comment (bowenliang123) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala:338]: This was to prevent changing the idents of following lines.

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala:338]: not a big deal, Git tools like GitHub could ignore space changes during diff view.
`()` is also fine if you insist.

Comment (bowenliang123) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala:338]: Thanks, let me keep it this time. The blame history is more valuable here for me.

**Comment by bowenliang123** [comment]
Thanks, merged to master(1.9.0) and branch-1.8 (1.8.1).

**Comment by bowenliang123** [comment]
Please consider propagating this change to other types of operations, if necessary.

**Pull request closed without merging**